### ISSUE TITLE
Missing support for LUKS cbc cipher

### ISSUE THREAD

**Issue by jarlethorsen** [hashcat/hashcat#1812]
I get the error "Invalid or unsupported LUKS cipher mode" when running mode 14600 on a LUKS with the following properties:

Version:       	1
Cipher name:   	aes
Cipher mode:   	cbc
Hash spec:     	sha256
Payload offset:	4096
MK bits:       	256
MK digest:     	b8 e1 c0 e8 a8 3d 80 1a 14 88 b2 f1 57 c5 23 ba 1e 13 66 f9 
MK salt:       	f7 dd 7f 24 94 bc cd a2 19 07 33 49 ea c3 07 82 
               	df 52 95 f9 77 cd df ef 78 90 7c 4d 27 5b 2e 7a 
MK iterations: 	28250
UUID:          	63fda6ee-248c-410d-95dd-62f4608141b6

Key Slot 0: ENABLED
	Iterations:         	115941
	Salt:               	78 b9 df e3 e8 cf dd 9c 59 67 20 70 53 b5 04 d2 
	                      	37 11 c8 e4 ce 56 df 0d 92 d6 57 eb 7b b8 ad c5 
	Key material offset:	8
	AF stripes:            	4000
Key Slot 1: DISABLED
Key Slot 2: DISABLED
Key Slot 3: DISABLED
Key Slot 4: DISABLED
Key Slot 5: DISABLED
Key Slot 6: DISABLED
Key Slot 7: DISABLED

I can provide the complete LUKS header if needed.

**Comment by philsmd** [comment]
Does this mean it is neither "cbc-plain64" nor "cbc-essiv" ? This is the cryptsetup luksDump output, right?

Are you able to open this volume with the correct password?

Are you able to create a similar (new) volume with cryptsetup? How do you create a volume with this cipher mode?

```
dd if=/dev/zero bs=1M count=32 of=test.img >/dev/null 2>/dev/null
cryptsetup luksFormat ./test.img -q --cipher aes-cbc
```

I get
```
Failed to setup dm-crypt key mapping for device ./test.img.
Check that kernel supports aes-cbc cipher (check syslog for more info).
```

I think it should be either cbc-plain64 or cbc-essiv. Can you somehow generate a image with the same Cipher name, Cipher mode, Hash spec as the luksDump from above?

**Comment by jarlethorsen** [comment]
> Does this mean it is neither "cbc-plain64" nor "cbc-essiv" ? This is the cryptsetup luksDump output, right?

Correct. This is the output of "cryptsetup luksDump". The first part of the header looks like this (notice the plain "cbc":

00000000: 4c55 4b53 babe 0001 6165 7300 0000 0000  LUKS....aes.....
00000010: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000020: 0000 0000 0000 0000 6362 6300 0000 0000  ........cbc.....
00000030: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000040: 0000 0000 0000 0000 7368 6132 3536 0000  ........sha256..

> Are you able to open this volume with the correct password?

I did not create the volume. Password unknown.

> Are you able to create a similar (new) volume with cryptsetup? How do you create a volume with this cipher mode?

I don't know. According to google there does not seem to be a "cbc" mode... Maybe this was made by an ancient version of cryptsetup and is in fact the same as cbc-plain?

**Comment by jsteube** [comment]
I've never seen "raw" cbc mode in LUKS before. The low iteration count also gives a hint. It could be a very old system or a very slow system (embedded) since LUKS meassures the time it takes to do N iterations in a second and then setting this as the iteration count. 

But I think it's just old. In that case it's possible that the algorithm is just not available anymore today, which is the reason why it's not in hashcat. While the kernel was developed, there was no such function to select from and therefore no one ever thought about having this.

**Comment by ombre8** [comment]
I just run into this Problem with a luks Volume created by [QNAPs QTS](https://www.qnap.com/qts/5.0/en-us/). I should recover a lost password for a Volume with this luks Header:

```
Version:       	1
Cipher name:   	aes
Cipher mode:   	cbc-plain
Hash spec:     	sha1
Payload offset:	4096
MK bits:       	256
MK digest:     	a8 a6 6c b5 e5 0c 3c 3e 4d ce 5c 50 b0 9e 97 1f 5c 74 14 26
MK salt:       	36 3e 91 3b 23 22 09 53 49 d2 c4 04 76 3c 2d 0b
               	ab 6e fd 55 78 81 d4 23 24 80 b3 d6 20 86 4d e9
MK iterations: 	3038
UUID:          	99c9ea48-b2ef-4741-8489-9c85e06629ef

Key Slot 0: ENABLED
	Iterations:         	12326
	Salt:               	f3 e0 17 a2 94 64 85 90 56 13 46 17 39 af 9b 35
	                      	3e ff 0c 9f 81 45 af 58 96 0c cd d4 a6 2f a9 cd
	Key material offset:	8
	AF stripes:            	4000
Key Slot 1: DISABLED
Key Slot 2: DISABLED
Key Slot 3: DISABLED
Key Slot 4: DISABLED
Key Slot 5: DISABLED
Key Slot 6: DISABLED
Key Slot 7: DISABLED
```
If I create a new Volume on said NAS it creates the Volume with `sha256` and not `sha1` anymore, but still aes-cbc-plain.
